students/john_rogers/lesson04/mailroom_pt2.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""
+mailroom.py -
+    1) prompt user for 3 actions - Send thank you, create report or quit
+Author: JohnR
+Version: 1.3
+Last updated: 12/29/2018
+Notes: TODO: create a single template using the .format(**d) method
+
+  v2.0 requirements:
+    1) use dict where appropriate
+    2) create a dict-based menu
+    3) create a single template using .format(**d) method for the letter
+    4) write a full set of letters to everyone on disk in individual files
+    5) add a function that goes through all donors, generates a thank you
+        letter, and writes it to disk as a text file.
+        A) use donor name and date for file name
+"""
+
+
+def main():
+    """
+    Create a list of donor and amounts, then get user input for the
+    various actions available.
+    :return: none
+    """
+
+    # make some global variables to experiment with
+    global db
+    global main_prompt
+    global main_dispatch
+
+    # create a main database
+    db = {'sting': {'d1': 13.45,
+                    'd2': 214.34,
+                    'd3': 453.23,
```

Comment:
what are d1, d2, and d3?

if you intend to use nested dict then use clear dict keys like "total", "donations" etc or simplify to just have donations as value:
`donors = {"bob": [1, 3, 4]}`